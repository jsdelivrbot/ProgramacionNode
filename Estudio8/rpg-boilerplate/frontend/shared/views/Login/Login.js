'use strict'
import React from 'react'
import LoginForm from '../../components/LoginForm/LoginForm'
import { HOME_ROUTE_PATH } from '../../routes/home'

import AppPublic from '../../templates/AppPublic/AppPublic'
import './Login.scss'

const Login = (props) => {
  return (
    <AppPublic noHeader>
      <LoginForm handleLoginSuccess={() => props.history.push(HOME_ROUTE_PATH)} />
    </AppPublic>
  )
}

export default Login
