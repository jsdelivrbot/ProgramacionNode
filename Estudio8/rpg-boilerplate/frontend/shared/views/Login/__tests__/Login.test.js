/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Login from '../Login'

describe('A </Login>', function () {
  it('should be selectable by class "login"', function () {
    expect(shallow(<Login />).is('.login')).toBe(true)
  })

  it('renders correctly', () => {
    expect(toJson(shallow(<Login />))).toMatchSnapshot()
  })

  it('renders correctly with input set', () => {
    const login = shallow(<Login />)
    login.find('#email').simulate('change', {target: {value: 'test@email.com'}})
    login.find('#password').simulate('change', {target: {value: 'test pass'}})

    expect(toJson(login)).toMatchSnapshot()
  })
})
