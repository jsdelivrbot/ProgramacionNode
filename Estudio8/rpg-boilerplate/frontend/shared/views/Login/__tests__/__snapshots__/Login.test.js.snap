exports[`A </Login> renders correctly 1`] = `
<withTheme(AppPublic)
  noHeader={true}
  query={[Function]}>
  <div
    className="login">
    <div
      className="page-title">
      Log in
    </div>
    <div
      className="signup row row-centered">
      Need an Evercast account? 
      <Link
        replace={false}
        to="/signup">
        Sign up now.
      </Link>
    </div>
    <withStyles(FormControl)
      className="row">
      <withStyles(InputLabel)
        htmlFor="login-input-label email">
        Email Address
      </withStyles(InputLabel)>
      <withStyles(Input)
        id="email"
        onChange={[Function]}
        value="" />
    </withStyles(FormControl)>
    <withStyles(FormControl)
      className="row">
      <withStyles(InputLabel)
        htmlFor="password login-input-label">
        Password
      </withStyles(InputLabel)>
      <withStyles(Input)
        id="password"
        onChange={[Function]}
        type="password"
        value="" />
    </withStyles(FormControl)>
    <div
      className="buttons-login">
      <withStyles(Button)
        className="row login-button"
        color="accent"
        onClick={[Function]}
        raised={true}>
        LOG IN
      </withStyles(Button)>
      <div
        className="row row-centered row-thin">
        OR
      </div>
      <withStyles(Button)
        className="dark row login-button"
        color="primary"
        raised={true}>
        LOG IN WITH GOOGLE
      </withStyles(Button)>
    </div>
    <div
      className="forgot row row-centered">
      <Link
        replace={false}
        to="/reset">
        Forgot password?
      </Link>
    </div>
  </div>
</withTheme(AppPublic)>
`;

exports[`A </Login> renders correctly with input set 1`] = `
<withTheme(AppPublic)
  noHeader={true}
  query={[Function]}>
  <div
    className="login">
    <div
      className="page-title">
      Log in
    </div>
    <div
      className="signup row row-centered">
      Need an Evercast account? 
      <Link
        replace={false}
        to="/signup">
        Sign up now.
      </Link>
    </div>
    <withStyles(FormControl)
      className="row">
      <withStyles(InputLabel)
        htmlFor="login-input-label email">
        Email Address
      </withStyles(InputLabel)>
      <withStyles(Input)
        id="email"
        onChange={[Function]}
        value="test@email.com" />
    </withStyles(FormControl)>
    <withStyles(FormControl)
      className="row">
      <withStyles(InputLabel)
        htmlFor="password login-input-label">
        Password
      </withStyles(InputLabel)>
      <withStyles(Input)
        id="password"
        onChange={[Function]}
        type="password"
        value="test pass" />
    </withStyles(FormControl)>
    <div
      className="buttons-login">
      <withStyles(Button)
        className="row login-button"
        color="accent"
        onClick={[Function]}
        raised={true}>
        LOG IN
      </withStyles(Button)>
      <div
        className="row row-centered row-thin">
        OR
      </div>
      <withStyles(Button)
        className="dark row login-button"
        color="primary"
        raised={true}>
        LOG IN WITH GOOGLE
      </withStyles(Button)>
    </div>
    <div
      className="forgot row row-centered">
      <Link
        replace={false}
        to="/reset">
        Forgot password?
      </Link>
    </div>
  </div>
</withTheme(AppPublic)>
`;
